### PR TITLE
CPHL can reference previously parsed highlights.

### LINKED ISSUE #1: Linking attributes "dynamically"
As we discussed on Gitter, it would be great if you could link colors "dynamically", so if you decide to change one, the subtypes of that color scheme would follow suit.

For example

``` vim
CPHL Function red - bold
CPHL pythonBuiltinFunc Function,bright Function Function,italic
```

So now `pythonBuiltinFunc` will always look like `Function` except that it will be slightly brighter and have italics.

### PR DESCRIPTION
**Pull request by tweekmonster**
Fixes #1 

Ping @tjdevries

### CODE REVIEW

**Comment by tjdevries** [comment]
I will test this out sometime tomorrow most likely.

**Comment by tjdevries** [comment]
So it seems to work for foreground, but I was hoping to be able to do something similar for background and style.

For example, this works just fine:

``` vim
CPHL Function blue,bright - bold
CPHL pythonBuiltinFunc Function,dark,dark - bold
```

It inherits the foreground color, and the modifies it. However, if I try to do something like this:

``` vim
CPHL Function blue,bright red bold
CPHL pythonBuiltinFunc Function,dark,dark Function
```

Then I get this error:

``` vim
Error detected while processing function colorpal#highlight:
line   66:
E421: Color name or number not recognized: ctermbg=cc6666 guibg=#cc6666
```

Where you thinking something different?

**Comment by tweekmonster** [comment]
It should work for all fields 😯  Must be a bug I missed.  I'll look into it.

**Comment by tjdevries** [comment]
It seems the for ctermbg, it doesn't have the `#` in front of it. I haven't had a chance to dig down and check if that's the case though.

At least that's what it seems the error says. Like I've said before, I have no idea how colors actually work in vim ;)

File: autoload/colorpal.vim [hunk 1]
```diff
@@ -1231,12 +1236,23 @@ function! colorpal#highlight(bang, group, ...) abort
...
+    endif
   else
     let style = ''
   endif
 
   if !empty(fg.bg.style)
+    if !has_key(s:parsed_hl, a:group)
+      let s:parsed_hl[a:group] = {
+            \ 'fg': [ctermfg, guifg],
+            \ 'bg': [guibg, guibg],
```
Comment (tweekmonster) [inline autoload/colorpal.vim]: @tjdevries 😑

**Comment by tjdevries** [comment]
Also, this appears to not be working either.

``` vim
CPHL pythonBuiltinFunc Function,dark,multiply(Operator,30)  - bold
```

I don't think I can do any of the "blend" operations, which I thought might provide interesting ideas. Like a python decorator could be a mix of function and object or something like that. Might just be more for novelty than anything else though :D

File: autoload/colorpal.vim (hunk 1 above)
Comment (tjdevries) [reply autoload/colorpal.vim]: Classic.

**Comment by tweekmonster** [comment]
> Like I've said before, I have no idea how colors actually work in vim ;)

`ctermbg` or `ctermfg` needs to be a named color or an integer 0-255.  I was accidentally storing the RGB hex triplet as the cterm color.

**Comment by tjdevries** [comment]
Gotcha. That makes sense.

**Comment by tjdevries** [comment]
I can confirm this works now: 

``` vim
CPHL pythonBuiltinFunc Function,dark Function,bright bold
```

**Comment by tjdevries** [comment]
I'm wondering if it'd be possible to do something like this:

``` vim
CPHL pythonBuiltinFunc Function,dark,multiply(Operator,30) Function,bright Function,italic

```

Currently I get this error:

``` vim
Error detected while processing function colorpal#highlight:
line   66:
E418: Illegal value: Function,italic
```

**Comment by tweekmonster** [comment]
I will see if I can do this real quick.

**Comment by tweekmonster** [comment]
It should work now.

**Comment by tjdevries** [comment]
It appears that this is working

``` vim
CPHL pythonBuiltinFunc Function,dark,multiply(Define,100) Function,add(Statement,5),bright,s+=90 Function,italic
```

I don't think I would actually use that exact one in my colorscheme.... but at least we know all the pieces are functioning to some extent :smile:

**Comment by tjdevries** [comment]
``` vim
" In the new colorscheme
CPHL Function blue,bright - bold
CPHL pythonBuiltinFunc Function,dark,dark Function Function
```

**Comment by tweekmonster** [comment]
I forgot to add support for multiple styles anyways.  I also want to support people who have terrible color coordination.

**Comment by tjdevries** [comment]
Well it looks like everything that I had wanted (at least that I could think of currently) is working. Thanks!

**Pull request merged**